src/main/scala/org/zalando/nakadi/client/Listener.scala

```diff
@@ -3,8 +3,8 @@ package org.zalando.nakadi.client
 
 trait Listener {
   def id: String
-  def onReceive(topic: String, partition: String, cursor: Cursor, event: Event)
-  def onConnectionOpened(topic: String, partition: String)
-  def onConnectionFailed(topic: String, partition: String, status: Int, error: String)
-  def onConnectionClosed(topic: String, partition: String, lastCursor: Option[Cursor])
+  def onReceive(topic: String, partition: String, cursor: Cursor, event: Event): Unit
+  def onConnectionOpened(topic: String, partition: String): Unit
+  def onConnectionFailed(topic: String, partition: String, status: Int, error: String): Unit
+  def onConnectionClosed(topic: String, partition: String, lastCursor: Option[Cursor]): Unit
```

Comment:
Leaving return values out gives a slightly different intention than explicitly setting them to `Unit`. I would set, to highlight that these don't return anything.

In public APIs there may be reason to leave the return values out, or use `Any`. I am not quite sure on that.